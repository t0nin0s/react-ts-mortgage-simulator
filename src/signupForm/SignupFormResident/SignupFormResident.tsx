import * as React from 'react';
import Card from '../../components/Card/Card';
import * as FontAwesome from 'react-fontawesome';
import MyButton from '../../components/Button/Button';
const styles = require('../signupForm.css');

interface SignupFormResidentProps {
  resident: string;
  handleOnChange: (event: React.MouseEvent<HTMLInputElement>) => void;
  handleNext: () => void;
}

class SignupFormResident extends React.Component<SignupFormResidentProps, {error: boolean}> {
  constructor() {
    super();
    this.state = {
      error: false
    };
  }

  validate = () => {
    if (this.props.resident) {
      this.setState({
        error: false
      });
      this.props.handleNext();
    } else {
      this.setState({
        error: true
      });
    }
  }

  render() {
    return (
      <Card
        formTitle="Sign up"
        icon="home"
      >
        <div className={styles.form__question}>
          Are you a UK resident?
        </div>
        <div className={styles.form__body}>
          <MyButton
            id="ukResident"
            value="No"
            text="No"
            icon="times-circle"
            color="red"
            handleOnChange={this.props.handleOnChange}
          />
          <MyButton
            id="ukResident"
            value="Yes"
            text="Yes"
            icon="check-circle"
            color="green"
            handleOnChange={this.props.handleOnChange}
          />
        </div>
        { this.state.error &&
          <div className={styles.error_message}>
            Please select first your residency...
          </div>
        }
        <div className={styles.form__submit}>
          <FontAwesome
            name="arrow-circle-right"
            size="3x"
            style={{ color: '#ECECEC' }}
            onClick={this.validate}
          />
        </div>
      </Card>
    );
  }
}

export default SignupFormResident;
