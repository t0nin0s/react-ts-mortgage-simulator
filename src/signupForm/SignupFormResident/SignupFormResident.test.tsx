import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import SignupFormResident from './SignupFormResident'

describe('SignupFormResident renders as expected', () => {
  it('should render as expected', () => {
    const component = shallow(
      <SignupFormResident
        resident=''
        handleOnChange={jest.fn()}
        handleNext={jest.fn()} />
    )
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
