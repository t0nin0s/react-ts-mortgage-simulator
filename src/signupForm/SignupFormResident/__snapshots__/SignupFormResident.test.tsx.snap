// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupFormResident renders as expected should render as expected 1`] = `
<Card
  formTitle="Sign up"
  icon="home"
>
  <div>
    Are you a UK resident?
  </div>
  <div>
    <MyButton
      color="red"
      handleOnChange={[Function]}
      icon="times-circle"
      id="ukResident"
      text="No"
      value="No"
    />
    <MyButton
      color="green"
      handleOnChange={[Function]}
      icon="check-circle"
      id="ukResident"
      text="Yes"
      value="Yes"
    />
  </div>
  <div>
    <FontAwesome
      name="arrow-circle-right"
      onClick={[Function]}
      size="3x"
      style={
        Object {
          "color": "#ECECEC",
        }
      }
    />
  </div>
</Card>
`;
